lib/shelley/test/unit/Cardano/Wallet/Shelley/NetworkSpec.hs

```diff
@@ -47,6 +54,54 @@ spec = describe "NetworkLayer regression test #1708" $ do
                     race_ (watchNodeTip nl (putMVar waiter)) (takeMVar waiter)
             void $ waitAnyCancel tasks
 
+    describe "Observer" $ do
+        describe "(query k) with typical use" $ beforeAll mockObserver $ do
+            -- Using monadic-property tests /just/ for the sake of testing with
+            -- multiple keys seem worthless.
```

Comment:
It'd be nice to test with multiple keys, since this is the main _point_ of this PR actually. Coming up with some properties would help testing possible edge-cases that aren't caught here, although may be a bit tricky here. At least, we should add two additional test scenarios:

- registering the same key twice
- registering multiple keys

Reply:
I should say, I’m much more worried about this causing further balance discrepancies than that it doesn’t handle registering the same key twice, or multiple keys, etc (though sure, good to test).